libtrip-service/src/main/java/com/mapbox/navigation/trip/service/MapboxTripService.kt

```diff
@@ -2,17 +2,56 @@ package com.mapbox.navigation.trip.service
 
 import com.mapbox.annotation.navigation.module.MapboxNavigationModule
 import com.mapbox.annotation.navigation.module.MapboxNavigationModuleType
+import com.mapbox.navigation.base.trip.MapboxNotificationData
+import com.mapbox.navigation.base.trip.RouteProgress
 import com.mapbox.navigation.base.trip.TripNotification
 import com.mapbox.navigation.base.trip.TripService
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.InternalCoroutinesApi
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import timber.log.Timber
 
+@InternalCoroutinesApi
+@ExperimentalCoroutinesApi
 @MapboxNavigationModule(MapboxNavigationModuleType.TripService, skipConfiguration = true)
-class MapboxTripService(override val tripNotification: TripNotification) : TripService {
+internal class MapboxTripService(
+    private val tripNotification: TripNotification,
+    private val callback: () -> Unit
+) : TripService {
 
-    override fun startService(stateListener: TripService.StateListener) {
-        TODO("not implemented")
+    private val serviceStarted = AtomicBoolean(false)
+
+    @InternalCoroutinesApi
+    override fun startService() {
+        if (!notificationDataChannel.isClosedForSend) {
+            notificationDataChannel.close()
+        }
+        notificationDataChannel = ConflatedBroadcastChannel()
+        when (serviceStarted.compareAndSet(false, true)) {
+            true -> {
+                callback()
+                notificationDataChannel.offer(
+                        MapboxNotificationData(tripNotification.getNotificationId(),
+                                tripNotification.getNotification())
+                )
+            }
+            false -> {
+                Timber.i("service already started")
```

Comment:
You can inject the modular `Logger` instance via the constructor and use it instead of hardcoding Timber.

Reply:
Yes, we could. But a better solution would be for the logger to expose a ConflatedBroadcastChannel. The logger would listen on that channel and any client could get access to that channel if it is exposed via a companion object from the logger. The client the would send messages to the logger tagged appropriately. The logger would print and/or do whatever it wants with the data: Sample code: I've simplified the logger class to a single function call "v()" for the sake of this discussion. Added an initializer and a function that returns a channel.  Added a data class to hold a the tag and message (MessageData).  Below the class, please see usage. You can make getLoggerChannel() a member of the Logger interface, if you like. Given this architecture, no injection is required and the Logger is already a singleton. 
object MapboxLogger : Logger {

    init {
        val subscription = MapboxLogger.sourceData.openSubscription()
        val context = CoroutineScope(Job() + Dispatchers.IO)
        context.launch {
            whileSelect {
                subscription.onReceiveOrClosed { channelState->
                    when (subscription.isClosedForReceive){
                        true -> {
                            false
                        }
                        false -> {
                            MapboxLogger.v(channelState.value.tag, channelState.value.msg)
                        }
                    }
                }
            }
        }
    }
    @LogLevel
    @Volatile
    var logLevel: Int = VERBOSE

    private val observer: AtomicReference<LoggerObserver> = AtomicReference()

    fun setObserver(observer: LoggerObserver) {
        this.observer.set(observer)
    }

    fun v(msg: Message) {
        v(null, msg, null)
    }
    data class MessageData constructor(val tag:String, val msg:String)
    private val sourceData:ConflatedBroadcastChannel<MessageData> = ConflatedBroadcastChannel()
    fun getLoggerChannel()   = sourceData
}
//Usage
MapboxLogger.getLoggerChannel().offer(MapboxLogger.MessageData("ThisIsMyTag", "Hello, wonderful world of channels")

Reply:
Besides @Guardiola31337's question, the idea sounds really good. One required change - the singleton that provides the communication you have in mind would have to have the actual Logger instance injected instead (or loaded in the `init` block by the module provider). That's because it can be replaced by the end developer.

Reply:
Also, the `getLoggerChannel` would have to be an interface method defined in `libnavigation-base` because other default modules do not depend on the SDK module, or the `Logger` module (as default logger can be replaced). This would require passing an instance via constructor anyway unless this singleton communication wrapper lives in the `libnavigation-utils` that all modules can depend on.

Reply:
All the above doesn't have to happen in this PR, we can follow up on logger setup later.